cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -39,12 +39,13 @@ def domain(request):
 
 
 @pytest.fixture(scope="module")
-def test_vm(setup_provider_modscope, provider, vm_name):
+def test_vm(setup_provider_modscope, provider, vm_name, request):
     """Fixture to provision appliance to the provider being tested if necessary"""
     vm = VM.factory(vm_name, provider, template_name=provider.data['small_template'])
 
     if not provider.mgmt.does_vm_exist(vm_name):
         vm.create_on_provider(find_in_cfme=True, allow_skip="default")
+        request.addfinalizer(vm.delete_from_provider)
```

Comment:
Great improvement to the test structure.

Consider a yield_fixture, where you can do more to wrap the delete in a try block and do things like log+continue if it fails, etc. Just a consideration, not necessary.